packages/context/src/watcher.ts

```diff
@@ -224,7 +226,18 @@ export const reactOnXmlFileChange = async (
     xmlUri: uri,
     changeType,
   });
+
   const documentPath = URI.parse(uri).fsPath;
+  if (isXMLView(documentPath)) {
+    if (
+      changeType === FileChangeType.Created ||
+      changeType === FileChangeType.Deleted
+    ) {
+      // reset cached view files
+      cache.setViewFiles(getWebappPath(documentPath), {});
```

Comment:
file has dependency to each other due to `id`, but yet we could invalidate a single file. 
Previous solution was to avoid micro programming. Furthermore these operations are not very often performed and building 105 xml files with roughly 240 lines of xml, takes less than a second. Anyway, now adapted to invalidate a single file.